libsolidity/codegen/ArrayUtils.cpp

```diff
@@ -289,10 +289,15 @@ void ArrayUtils::copyArrayToStorage(ArrayType const& _targetType, ArrayType cons
 			// stack: target_ref target_data_end source_data_pos target_data_pos_updated source_data_end
 			_context << Instruction::POP << Instruction::SWAP1 << Instruction::POP;
 			// stack: target_ref target_data_end target_data_pos_updated
-			if (targetBaseType->storageBytes() < 32)
-				utils.clearStorageLoop(TypeProvider::uint256());
+			if (_targetType.isDynamicallySized())
+				ArrayUtils(_context).clearDynamicArray(_targetType);
 			else
-				utils.clearStorageLoop(targetBaseType);
+			{
+				if (targetBaseType->storageBytes() < 32)
+					utils.clearStorageLoop(TypeProvider::uint256(), /* _canOverflow */ true);
+				else
+					utils.clearStorageLoop(targetBaseType, /* _canOverflow */ true);
+			}
```

Comment:
Why does this change that much? Seems a bit fishy to me - the stack is not even in the right state for ``clearDynamicArray`` is it?

Reply:
Yeah, it was indeed wrong. Fixed